session/session.go

```diff
@@ -1494,14 +1502,18 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	if err := executor.ResetContextOfStmt(s, stmtNode); err != nil {
 		return nil, err
 	}
+	normalizedSQL, digest := s.sessionVars.StmtCtx.SQLDigest()
+	if variable.TopSQLEnabled() {
```

Comment:
maybe we can move `variable.TopSQLEnabled()` logic into `AttachSQLInfo`

Reply:
I prefer not. If `variable.TopSQLEnabled()` is true, we need some other process to get sql_digest and plan_digest.

Put this check logic outside can avoid those process cost.